src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebProxy.cs

```diff
@@ -18,6 +18,34 @@ internal WebProxy(Uri address)
             _proxyAddress = address;
         }
 
+        public override bool Equals(object obj)
+        {
+            if (obj == null || GetType() != obj.GetType())
+            {
+                return false;
+            }
+            
+            return Equals(obj as WebProxy);
+        }
+
+        public override int GetHashCode()
```

Comment:
```suggestion
        public override int GetHashCode() => _proxyAddress.GetHashCode() * 17 + _credentials.GetHashCode();
```

Reply:
Updated, but was forced to add additional parenthesis by the style checker.